cli/initdata/initdata.go

```diff
@@ -0,0 +1,212 @@
+// Copyright 2025 Edgeless Systems GmbH
+// SPDX-License-Identifier: BUSL-1.1
+
+package initdata
+
+import (
+	"bytes"
+	"compress/gzip"
+	"crypto/sha256"
+	"crypto/sha512"
+	"encoding/base64"
+	"fmt"
+	"io"
+
+	"github.com/pelletier/go-toml/v2"
+)
+
+const (
+	// InitdataVersion is the sole supported version of Initdata.
+	InitdataVersion = "0.1.0"
+
+	// InitdataAnnotationKey as specified in: https://github.com/kata-containers/kata-containers/blob/f6ff9cf7176989d414bf3f45a5b0c0b9fdb1bf3a/src/libs/kata-types/src/annotations/mod.rs#L276
+	InitdataAnnotationKey = "io.katacontainers.config.hypervisor.cc_init_data"
+)
+
+// Initdata follows the standardized structure format as described here:
+// https://github.com/confidential-containers/trustee/blob/d9c0b6fa01a042052ba93c703b1ff6aab2a2b63f/kbs/docs/initdata.md?plain=1#L82-L88.
+type Initdata struct {
+	version   string
+	algorithm string
+	data      map[string]string
+}
+
+// New creates a new Initdata struct and validates it.
+func New(algorithm string, data map[string]string) (*Initdata, error) {
+	i := Initdata{
+		version:   InitdataVersion,
+		algorithm: algorithm,
+		data:      data,
+	}
+
+	if err := i.Validate(); err != nil {
+		return nil, fmt.Errorf("validating initdata: %w", err)
+	}
+
+	return &i, nil
+}
+
+// Validate the parsed initdata.
+func (i *Initdata) Validate() error {
+	if i.version != InitdataVersion {
+		return fmt.Errorf(
+			"specified initdata version is %s, but supported is only version %s",
+			i.version,
+			InitdataVersion,
+		)
+	}
+
+	// Attempt to calculate the Digest of an empty array.
+	// Fails with an err if the Initdata was created with an unsupported algorithm.
+	if _, err := i.Digest([]byte{}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Digest calculates the hash of the Initdata.
+func (i *Initdata) Digest(digestible []byte) (string, error) {
```

Comment:
Do you think this needs to be exported? The interface is a bit odd, so I'd rather keep it internal.